public/app/features/alerting/unified/components/silences/SilencesTable.tsx

```diff
@@ -70,24 +74,25 @@ const SilencesTable: FC<Props> = ({ silences, alertManagerAlerts, alertManagerSo
               </Link>
             </div>
           )}
-          {!!items.length ? (
-            <>
-              <DynamicTable
-                items={items}
-                cols={columns}
-                isExpandable
-                renderExpandedContent={({ data }) => <SilenceDetails silence={data} />}
-              />
-              {showExpiredSilencesBanner && (
-                <div className={styles.callout}>
-                  <Icon className={styles.calloutIcon} name="info-circle" />
-                  <span>Expired silences are automatically deleted after 5 days.</span>
-                </div>
-              )}
-            </>
-          ) : (
-            'No matching silences found'
-          )}
+          {hasReadPermissions &&
```

Comment:
I wonder if we need this check. If a user does not have the `AlertingInstanceRead` permission they should not see that page at all. 
If we want to keep that check here I think we should also add permission checking when loading silences in the redux action